Initial Layout

ernestognw opened the pull request:
Project dashboard layout was added. 

Although I tried to follow the Javascript style guides from Google, I had some questions regarding the following:

1. The docs about import statements are not pretty clear since there are destructured imports, renamed imports, imports with renames, etc. Probably I'm ignoring something, but the criteria I followed was:

    - First, default imports alphabetically
    - Second, destructured imports (with destructured alphabetized)

    Hope to get some opinions about this.

2. The JSDoc requires me to write the `@return` and `@param` in every React component. This is a pattern that we'll see a lot, and I have some comments about it

    - `@return` is always typed the same. Since we're returning a React Node, I think it is pretty clear when we're writing a component, so I see no `@return` need. I just want to know what you think.
    - Every React component is required by ESLint to have their `PropTypes` which is the internal spec for the params received, so when writing `@param`'s, this info is duplicated. That's why, again, I think writing JSDoc here is unnecessary since React PropTypes are more standard.

### General updates
- **[Material UI](https://material-ui.com/) Setup**: This is the UI library we agreed to use
- **[Styled Components Setup](https://styled-components.com/)**: This is a css-in-js tool to write dynamic and custom components
- **[React Router](https://reactrouter.com/web/guides/quick-start) Setup**: Library to handle client-side routing
- **Layout added**: Basic navbar and routes in drawer were added
- **Basic routes**: About and mixed content routes were added.
- **Routes Tests**
- **Layout Tests**
- **App Tests**

**Note**: Please review and approve #1 first

## Screenshots
# ![Screenshot 1](https://i.ibb.co/GMmcsfD/enamel-step-2020-appspot-com-about-2.png)
# ![Screenshot 2](https://i.ibb.co/zm1fyzS/enamel-step-2020-appspot-com-about-1.png)

JonathanChavezTamales commented:
Great! LGTM too

livvielin commented on public/css/main.css:
```diff
@@ -0,0 +1,7 @@
+/* Links style reset */
+body a {
+  display: contents;
```
The declarations should be alphabetized.

ernestognw replied:
Done

ernestognw commented:
> Haven't finished reviewing yet, but would you be able to specify which things changed from https://github.com/googleinterns/step180-2020/pull/1/files (since that one isn't merged yet) and maybe provide a screenshot? These can help with reviewing. Thanks!

Np! I've solved the comments on #1 and updated this Pull Request, now you're able to only see the current changes.
Also, I added some screenshots in the description.

lmk if you need anything else.

livvielin commented on src/views/about/index.js:
```diff
@@ -0,0 +1,20 @@
+ *
+ * This component should have everything to be rendered
+ * under /about route. It is intended to have a description
+ * about the data and what we're doing with the app
```
nit: We generally try to avoid using "we" in comments. I would also recommend just saying what this component does rather than the intention. Maybe rephrase to something like "It contains a description of the data and the purpose of the app."

ernestognw replied:
Awesome, I readapted almost every comment to avoid "we" and start with a verb

livvielin commented on src/utils/test-utils.js:
```diff
@@ -0,0 +1,58 @@
+};
+
+/**
+ * Custom render is an adaptatoion of original render
```
nit: adaptation

ernestognw replied:
Done!

livvielin commented on src/utils/test-utils.js:
```diff
@@ -0,0 +1,58 @@
+import {render} from '@testing-library/react';
+
+/**
+ * This providers are used to setup context needed
```
nit: These providers

ernestognw replied:
Done!

livvielin commented on src/utils/test-utils.js:
```diff
@@ -0,0 +1,58 @@
+
+/**
+ * Custom render is an adaptatoion of original render
+ * with context providers to get the tests work.
```
nit: tests **to** work

ernestognw replied:
Done, also corrected the same problem at the top of the document

livvielin commented on src/utils/test-utils.js:
```diff
@@ -0,0 +1,58 @@
+ * In this file, original test functions and utils are
+ * overridden to provide some customizations like a
+ * custom render with every needed wrapper to get
+ * the testings work
```
nit: the tests to work

ernestognw replied:
Done!

livvielin commented on src/config/theme/index.js:
```diff
@@ -0,0 +1,21 @@
+  // Add special theming using this reference: https://material-ui.com/customization/default-theme/
+  palette: {
+    primary: {
+      light: '#63ccff',
```
nit: either put these in alphabetical order, or an intuitive order? I think main, light, dark, contrast makes sense to me, but feel free to not do this if it already makes sense in current order.

livvielin commented on src/layouts/main/styles/functions.js:
```diff
@@ -0,0 +1,74 @@
+    duration: theme.transitions.duration.leavingScreen,
+  })};
+  overflow-x: hidden;
+  width: ${theme.spacing(9) + 1}px;
```
Curious why these numbers were chosen and why adding 1?

ernestognw replied:
tbh, I copied some style examples from the official docs, and adapted accordingly, I removed the + 1 since it actually has no purpose.

Regarding the spacing, the spec is [here](https://material-ui.com/customization/spacing/), but there is no real motivation to select `theme.spacing(9)`, just an opinionated decision from my side.

Let me know if you have a clue on how we can organize these decisions better.

livvielin commented on src/layouts/main/styles/functions.js:
```diff
@@ -0,0 +1,74 @@
+import {drawerWidth} from './constants';
+
+/**
+ * To return drawer styles when is open.
```
nit: For comments, usually we like to start with a verb. e.g. "Returns drawer styles..."

For here and in other places.

ernestognw replied:
Done, also I rewrote almost every comment to try to fit into this recommendation.

livvielin commented on src/layouts/main/styles/functions.js:
```diff
@@ -0,0 +1,74 @@
+ * @param {{open: boolean, theme: object}} props
+ * @return {css} styles according to open state
+ */
+const drawerToggles = ({open, theme}) => {
```
Maybe name this toggleDrawer instead? Tend to like the verb since we're doing something with this function.

ernestognw replied:
I decided to name it `getToggleDrawerStyles`. I think `toggleDrawer` sounds more like a function to handle component functionality. I just don't want to create confusion with the purpose of these functions.

Let me know if it sounds better, I was thinking about it for a few minutes

livvielin commented on src/layouts/main/styles/functions.js:
```diff
@@ -0,0 +1,74 @@
+ * @param {{active: boolean, theme: object}} props
+ * @return {css} styles according to ac state
+ */
+const tabToggle = ({active, theme}) => {
```
I'm not sure I understand the function naming for this? Also, would it be possible to merge this function with getActiveTabStyles?

ernestognw replied:
Some context: I had some problems with the css-in-js, since it uses indentation inside JS template literals, and the linter shows an indentation error. That's why I decided to put these styles in subfunctions since it respects the indentation. I'm attaching an SS for that:

![ESLint Error](https://i.ibb.co/PTd7xTf/Captura-de-Pantalla-2020-08-03-a-la-s-0-24-50.png)
![Original function](https://i.ibb.co/mCkcFX2/Captura-de-Pantalla-2020-08-03-a-la-s-0-26-10.png)

That been said, I tried to use only `getActiveTabStyles` as suggested, and it worked, so I replaced it 😄 , I'm just saying this just to show that we might have to add a special rule in the ESLint configuration to avoid indentation errors. In that case, I can simplify the way we use these functions, which is important since it'll be frequent to have them.

ernestognw commented on .gitignore:
```diff
@@ -24,4 +24,8 @@ yarn-debug.log*
 yarn-error.log*
 
 # credentials
+gcloud_creds.json
```
Added this in advance to avoid future accidents, since @JonathanChavezTamales and @SofiaVega are using this file to handle credentials.

I noticed some branches already have this, but as early as we get this to master branch, the better.

livvielin replied:
lgtm

livvielin commented on src/layouts/main/styles/functions.js:
```diff
@@ -0,0 +1,68 @@
+ * @param {{open: boolean, theme: object}} props
+ * @return {css} styles according to open state
+ */
+const getToggleDrawerStyles = ({open, theme}) => {
```
nit: maybe just getDrawerStyles? Since you're not actually toggling

ernestognw replied:
Done!

livvielin commented on src/utils/test-utils.js:
```diff
@@ -0,0 +1,56 @@
+ *    children is the component to be rendered in tests
+ * @return {ReactNode} Enhanced children with providers
+ */
+const AllTheProviders = ({children}) => {
```
nit: maybe just AllProviders, since "The" doesn't seem to be necessary?

ernestognw replied:
Done!
